### DISCUSSION TITLE
using color blocks as button response

### DISCUSSION THREAD

**Discussion by ZifengLIU143** [jspsych/jsPsych#2799] [Q&A] [answered]
Hi, 

I am trying to use color blocks as button response. That means the button is padding with different colors without any characters. 
I tried to do this by modifying the css element <style> for each button, but I cannot get rid of the white rectangle. Does anyone know how to solve this or have any better idea?

```
  <style>
      #jspsych-html-button-response-button-0 {
        background-color: brown;
        border-radius: 84px
        }
  </style>
```
The output looks like this:

<img width="229" alt="Capture d’écran 2022-10-03 à 11 23 36" src="https://user-images.githubusercontent.com/57791409/193543772-ac9fa318-ef44-4fa0-9105-8a49a3ddd7ad.png">

Best, 

Zifeng

**Comment by ChristopheBossens** [comment, marked as answer]
I would take advantage of the fact that the html-button-response plugin has a `button_html` atttribute in which you can customize your buttons, in combination with defining some custom CSSproperties (as you are already doing)

CSS Styles:
```
    <style>
        .color-btn {
            height: 32px;
            width: 32px;
            cursor: pointer;
            border-radius: 2px;
            border-width: 0px;
        }

        .color-btn-0 {
            background-color: red;
        }

        .color-btn-1 {
            background-color: green;
        }

        .color-btn-2 {
            background-color: blue;
        }

    </style>
```
Definition of the jsPsychHtmlButtonResponse trial:

```
    const button_trial = {
        type: jsPsychHtmlButtonResponse,
        stimulus: '<p style="font-size:48px; color:red;">GREEN</p>',
        button_html: [
            '<button class="color-btn color-btn-0">%choice%</button>', 
            '<button class="color-btn color-btn-1">%choice%</button>', 
            '<button class="color-btn color-btn-2">%choice%</button>'
        ],
        choices: ['', '', ''],
        prompt: "<p>What color is the ink?</p>",
    }
```

Output

![color_buttons](https://user-images.githubusercontent.com/7238485/194001164-ac006984-2b9f-4007-b7f4-60f4fa22d530.png)


Is this what you are looking for?

**Reply by ZifengLIU143** [reply]
Cool! It is exactly what I want! Thank you very much Christophe :)
